src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/Acquisition/MissingSetupComponentRegistrationService.cs

```diff
@@ -81,7 +81,7 @@ internal class MissingSetupComponentRegistrationService : OnceInitializedOnceDis
         {
             if (_projectPathToProjectConfigurationsMap is null)
             {
-                Interlocked.CompareExchange(ref _projectPathToProjectConfigurationsMap, new(), null);
+                Interlocked.CompareExchange(ref _projectPathToProjectConfigurationsMap, new(StringComparers.Paths), null);
```

Comment:
I often wonder if we should introduce a `FilePath` type as a thin wrapper around `string` to prevent issues like this.

Reply:
I'd love a zero cost abstraction in the language for this kind of thing.